### PR TITLE
 cmake: link libcommon with libstdc++ statically if WITH_STATIC_LIBSTDCXX

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
i tested the change by 

1. compiling ceph-mon on a xenial with gcc-7 installed, `-DWITH_STATIC_LIBSTDCXX=ON`,
2. the `ceph-mon --version` runs fine on the building host, and on a xenial env where gcc-7 is not installed.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -743,7 +745,7 @@ if (WITH_MGR)
   add_executable(ceph-mgr ${mgr_srcs}
                  $<TARGET_OBJECTS:heap_profiler_objs>)
   target_include_directories(ceph-mgr SYSTEM PRIVATE "${PYTHON_INCLUDE_DIRS}")
-  target_link_libraries(ceph-mgr osdc client global-static ceph-common
+  target_link_libraries(ceph-mgr osdc client global-static common
```
Comment (cbodley) [inline src/CMakeLists.txt]: given `target_link_libraries(global-static common)` in global/CMakeLists.txt, is it necessary to list common for these executables?

Comment (tchaikov) [reply src/CMakeLists.txt]: technically no, but we cannot rely on this fact that `global-static` depends on `common`.  strictly speaking, it's the internals of `global-static`.

**Comment by yuriw** [comment]
[wip-yuri-testing-2017-12-12-1602](https://trello.com/c/a1IyuU7P/210-wip-yuri-testing-2017-12-12-1602)

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-12-15_11:03:39-rados-wip-kefu-testing-2017-12-15-1528-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-12-16_03:28:12-rados-wip-kefu-testing-2017-12-15-1528-distro-basic-mira/

**Comment by batrick** [comment]
@tchaikov I was hoping this would resolve similar errors in the fs suite but apparantly it did not. Do you think these are a different problem?

/ceph/teuthology-archive/pdonnell-2018-01-08_23:36:57-fs-wip-pdonnell-testing-20180105.234510-testing-basic-smithi/2044328/remote/smithi044/log/valgrind/client.0.log.gz

**Comment by tchaikov** [comment]
@batrick it's a regression introduced by 75c5e5cc26ae202bc0560fdfc38a3ecce1188ad7. will post a fix for it.

**Comment by tchaikov** [comment]
@batrick #19892

**Pull request merged**